### DISCUSSION TITLE
rTorrent user can't access external program (php)

### DISCUSSION THREAD

**Discussion by chapit-nicolas** [Novik/ruTorrent#2530] [Q&A] [answered]
Hi there,

I did a fresh install with the rtInst script a month (or 2) ago. At that time, it was rTorrent 0.9.8 and RuTorrent 3.x.x the latest.

I always have multiple error messages regarding `rTorrent user can't access external program (php)`  or about pgrep, curl, python and ffmpeg. So recently, I saw that rtInst updated their script to use RuTorrent 4.x, so I re-use it to install this new version, in hope it will fix those error messages. But it did not...

So I did many checks in configs:
- does the ruTorrent's user has access on lib: (which php or php -v) It works
- does the ruTorrent's config has the path: yes `$pathToExternals` is set, except for ffmpeg
- which user runs rTorrent and RuTorrent: `rtorrent`, because this is the only one set and it uses session/download paths I set in the .rc config file

I'm sure it's a silly issue, but I can't find the solution... I saw many posts about that, I tried some fixes proposed, but nothing...
In the error message, it says rTorrent user, how can I be sure of which user is used? In the auto-start script, User has `rtorrent` for value, and it matches my system user.

Everything seems to work, because I can add torrents, they are downloaded and seeded

Current versions installed:
 - Ubuntu: 22.04
 - Nginx: 1.18
 - PHP (cli & fpm): 8.2.7
 - RuTorrent: 4.1.6
 - rTorrent: 0.9.8
 
 Please help me :)
 Thanks in advance

**Comment by stickz** [comment, marked as answer]
The rtinst install script is broken. There are multiple bugs - not just this one with php and the lack of support for ruTorrent version 4. It was not updated to support V4. Here are some alternative install scripts for ruTorrent. I use the swizzin one. https://github.com/Novik/ruTorrent/discussions/2482

**Comment by chapit-nicolas** [comment]
Oh ok... I look the Swizzin's script, and it seems it installs everything including the OS. I don't need to install the OS. The machine where I installed rtorrent and rutorrent is not a seedbox. I guess, i will have to redo the install manually...

About the rtInst, I built a test VM with Ubuntu 22.04, and I used the script with RuTorrent 4.1.6, and I was able to launch the RuTorrent without any errors. I don't have access on this VM right now, but once I will have it, I will do a full check of this install.

Thanks @stickz .
